### ISSUE TITLE
Non-optimal model cannot be asked for their `termination_condition`

### ISSUE THREAD

**Issue by timtroendle** [calliope-project/calliope#365]
## Problem description

When running Calliope programmatically and the model run was infeasible or suboptimal, there is no proper way anymore to know that is was not optimal. Before, one would ask for `model.results.termination_condition` after the run finished which would return `infeasible` or `suboptimal`. That is not possible anymore.

I believe this is a regression introduced in ce30b08f5288a3ac8484a0999ac0891c294b7fee.

It should either be possible again to ask for the termination condition, or there should be an Exception (actually I think the latter makes even more sense).

## Steps to reproduce the problem

```Python
model = Calliope(path_to_model) # make sure it's infeasible
model.run()
print(model.results.termination_condition) # will fail
print(model.inputs.termination_condition) # will work but is not logical
print(model._model_data.termination_condition) # will work but is private
```

## Calliope version

0.6.7.dev

**Comment by timtroendle** [comment]
I just stumbled upon the fact that `SPORES` mode has a similar issue: when the first run terminates non-optimal, the SPORES mode fails with an obscure error. The problem is it tried to access `model.result` which does not exist. To me, this supports the idea of throwing an exception. If you really want to catch the exception (like in a CLI `calliope run` call), you can always do. But the default would be a failure -- instead of the other way round as it is now.

**Comment by brynpickering** [comment]
Fixed as of d3413ffd7f0f7712a4f7ef8dace7682d722e1630

**Comment by brynpickering** [comment]
Just to note: an exception is a bad idea imo because then you lose `backend_model`. This is partly related to #314, where we discussed building the `backend_model` as a completely independent step from running it. As it is, they are both done at once, so such an exception would not allow a super-user to debug the pyomo model object for possible reasons for the infeasibility.
